src/DynamoCore/Graph/Nodes/DummyNode.cs

```diff
@@ -398,5 +404,75 @@ private XmlElement OriginalXmlNodeContent
                 return originalXmlElement;
             }
         }
+
+        /// <summary>
+        /// This will return the NodeModel for a dummy node. 
+        /// </summary>
+        /// <param name="json"></param>
+        /// <param name="libraryServices"></param>
+        /// <param name="factory"></param>
+        /// <param name="isTestMode"></param>
+        /// <param name="manager"></param>
+        internal NodeModel GetNodeModelForDummyNode(string json, LibraryServices libraryServices,
+                                                  NodeFactory factory, bool isTestMode, CustomNodeManager manager)
+        {
+            var settings = new JsonSerializerSettings
+            {
+                Error = (sender, args) =>
+                {
+                    args.ErrorContext.Handled = true;
+                    Console.WriteLine(args.ErrorContext.Error);
+                },
+                ReferenceLoopHandling = ReferenceLoopHandling.Ignore,
+                TypeNameHandling = TypeNameHandling.Auto,
+                Formatting = Newtonsoft.Json.Formatting.Indented,
+                Culture = CultureInfo.InvariantCulture,
+                Converters = new List<JsonConverter>{
+                        new NodeReadConverter(manager, libraryServices, factory, isTestMode),
+                        new TypedParameterConverter()
+                    },
+                ReferenceResolverProvider = () => { return new IdReferenceResolver(); }
+            };
+
+            var result = SerializationExtensions.ReplaceTypeDeclarations(json, true);
+            var resolvedNodeModel = JsonConvert.DeserializeObject<NodeModel>(result, settings);
+
+            // If the resolved node model is not a dummy node, then copy the node view properties from the dummy node to the resolved version of that node. 
+            if (!(resolvedNodeModel is DummyNode))
+            {
+                SetNodeViewDataOnResolvedNode(this, resolvedNodeModel);
+            }
+            else
+            {
+                this.Log(string.Format("This graph has a node with id:{0} and name:{1}, but it could not be resolved",
+                                       resolvedNodeModel.GUID, resolvedNodeModel.Name)
+                                       , WarningLevel.Moderate);
+            }
+
+            return resolvedNodeModel;
+        }
+
+        /// <summary>
+        /// This will set the dummy node's node view properties to the resolved node
+        /// </summary>
+        /// <param name="dummyNode"></param>
+        /// <param name="resolvedNode"></param>
+        private void SetNodeViewDataOnResolvedNode(NodeModel dummyNode, NodeModel resolvedNode)
+        {
+            if (dummyNode == null || resolvedNode == null)
+            {
+                return;
+            }
+
+            resolvedNode.X = dummyNode.X;
```

Comment:
one thought I have about this is - is there a way to ensure we are transferring all view properties of the dummyNode - so we don't forget to update this if new view properties are added? Something like checking the ExtraNodeViewInfo object for all properties or something? This is not required - but something to think about.